pkg/privilege/privileges/privileges.go

```diff
@@ -78,13 +78,18 @@ type UserPrivileges struct {
 	host string
 	*Handle
 	extensionAccessCheckFuncs []extension.AccessCheckFunc
+	authPlugins               map[string]*extension.AuthPlugin
```

Comment:
Functions like `IsValidHash` and `GetAuthPluginForConnection` will need this reference to load the plugins. Would prefer we keep this here since we load the extensions when creating `UserPrivileges` anyways.

Reply:
Another option is to move it to `Handle` which is included in line 79. Seems it is shared by all connections and is used to cache some global meta.

Reply:
It might be better to leave it as is to be consistent with `extensionAccessCheckFuncs`. Any concerns on keeping this inside `UserPrivileges`?